### PR TITLE
refactor(sqlalchemy): remove unnecessary `sa.text` calls

### PR DESCRIPTION
**Pull request by cpcloud**
This PR replaces superfluous `sa.text` calls with direct string execution where possible. This is already what SQLAlchemy does, here are just removing one unnecessary function call.

Depends on #5282.

### CODE REVIEW

**Comment by kszucs** [comment]
@cpcloud it needs a rebase

**Comment by cpcloud** [comment]
> @cpcloud it needs a rebase

@kszucs Done!

**Comment by cpcloud** [comment]
The trino tests should be less flaky after #5303. I'll rebase.

File: ibis/backends/mysql/__init__.py [hunk 1]
```diff
@@ -106,19 +106,18 @@ def do_connect(
...
 
             yield bind
+            stmt = sa.text("SET @@session.time_zone = :prev").bindparams(prev=prev)
             try:
-                bind.execute(
-                    sa.text("SET @@session.time_zone = :prev").bindparams(prev=prev)
-                )
+                bind.execute(stmt)
             except Exception as e:  # noqa: BLE001
-                warnings.warn(sa.text(f"Couldn't reset MySQL timezone: {e}"))
```
Comment (kszucs) [inline ibis/backends/mysql/__init__.py:121]: Wow! :)

**Review by kszucs — APPROVED** [review]
LGTM

**Pull request merged**